src/random_graph.rs

```diff
@@ -380,6 +380,64 @@ pub fn random_geometric_graph(
     Ok(graph)
 }
 
+/// Return a :math:`\mathbb{H}^2` hyperbolic random undirected graph.
+///
+/// The :math:`\mathbb{H}^2` hyperbolic random graph model connects pairs of nodes with probability
+///
+/// .. math::
+///
+///     P[(u,v) \in E] = \frac{1}{1+\exp(\beta(d(u,v) - R)/2)},
+///
+/// a function that decreases with the hyperbolic distance between nodes :math:`u` and :math:`v`
+///
+/// .. math::
+///
+///     d(u,v) = \text{arccosh}\left[\cosh(r_u) \cosh(r_v) - \sinh(r_u) \sinh(r_v)\cos(\theta_u-\theta_v)\right],
+///
+/// where :math:`r_u` and :math:`\theta_u` are the hyperbolic polar coordinates of node :math:`u`.
+///
+/// The number of nodes is inferred from the coordinates ``radii`` and ``angles``. ``radii`` and
+/// ``angles`` must have the same size and cannot be empty. If ``beta`` is ``None``, all pairs of
+/// nodes with a distance smaller than `r` are connected.
+///
+/// D. Krioukov et al. "Hyperbolic geometry of complex networks", Phys. Rev. E 82, pp 036106, 2010.
+///
+/// This algorithm has a time complexity of :math:`O(n^2)`.
+///
+/// :param list[float] radii: radial coordinates (nonnegative) of the nodes.
+/// :param list[float] angles: angular coordinates (in :math:`[-\pi, \pi]`) of the nodes.
+/// :param float beta: Sigmoid sharpness (nonnegative) of the connection probability.
+/// :param float r: Distance at which the connection probability is 0.5 for the probabilistic model.
+///     Threshold when `beta` is `None`.
+/// :param int seed: An optional seed to use for the random number generator.
+///
+/// :return: A PyGraph object
+/// :rtype: PyGraph
+#[pyfunction]
+#[pyo3(text_signature = "(radii, angles, r, beta, /, seed=None)")]
+pub fn undirected_hyperbolic_random_graph(
```

Comment:
If literature always assumes a Hyperbolic Random Graph is undirected, you can shorten the name of this function to `undirected_hyperbolic_random_graph`